client/landing/jetpack-cloud/components/threat-item/utils.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Threat, ThreatFix, ThreatType } from './types';
+
+export function getThreatType( threat: Threat ): ThreatType {
+	if ( threat.hasOwnProperty( 'diff' ) ) {
+		return 'core';
+	}
+
+	if ( threat.hasOwnProperty( 'context' ) ) {
+		return 'file';
+	}
+
+	if ( threat.hasOwnProperty( 'extension' ) ) {
+		// 'plugin' or 'theme'
+		const { extension = { type: 'unknown' } } = threat;
+		return extension.type;
+	}
+
+	if ( threat.hasOwnProperty( 'rows' ) ) {
+		return 'database';
+	}
+
+	if ( 'Suspicious.Links' === threat.signature ) {
+		return 'database';
+	}
+
+	return 'none';
+}
+
+export const getThreatVulnerability = ( threat: Threat ): string | i18nCalypso.TranslateResult => {
+	switch ( getThreatType( threat ) ) {
+		case 'core':
+			return translate( 'Vulnerability found in WordPress' );
+
+		case 'file':
+			return translate( 'Threat found %(signature)s', {
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Threat found ({{signature/}})' )` *ES Score: 7*